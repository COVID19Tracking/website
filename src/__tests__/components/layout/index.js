import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery, StaticQuery } from 'gatsby'
import Layout from '../../../components/layout/index'

beforeEach(() => {
  StaticQuery.mockImplementation(({ render }) =>
    render({
      site: {
        siteMetadata: {
          production: true,
        },
      },
    }),
  )

  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        title: 'Test site',
      },
    },
    allNavigationYaml: {
      edges: [
        {
          node: {
            items: [
              {
                link: '/test-a',
                title: 'Test A',
              },
              {
                link: '/test-b',
                title: 'Test B',
              },
            ],
          },
        },
      ],
    },
  }))
})

describe('Components : Layout : Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <Layout title="Sample title">
            <p>Content</p>
          </Layout>
        </>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const textHeavytree = renderer
      .create(
        <>
          <Layout title="Sample title" textHeavy narrow>
            <p>Content</p>
          </Layout>
        </>,
      )
      .toJSON()
    expect(textHeavytree).toMatchSnapshot()
  })
})
