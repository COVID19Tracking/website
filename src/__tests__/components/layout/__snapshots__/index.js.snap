// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Layout : Header renders correctly 1`] = `
Array [
  <a
    data-reach-skip-link=""
    data-reach-skip-nav-link=""
    href="#reach-skip-nav"
  >
    Skip site navigation
  </a>,
  <header
    className="site-header siteHeader  "
  >
    <div
      className="container container showBackground "
    >
      <div
        className=""
        style={
          Object {
            "height": 0,
            "opacity": 0,
            "overflow": "hidden",
            "transition": "height 500ms ease-in-out,opacity 500ms ease-in-out,background 500ms ease-in-out",
          }
        }
      >
        <div
          className="mobileMenu"
        >
          <div
            className="searchInput"
          >
            <form
              action="/search"
              method="get"
              onSubmit={[Function]}
            >
              <input
                autoComplete="off"
                name="q"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Search"
                type="search"
              />
            </form>
            <button
              className="searchSubmit"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                aria-hidden="true"
                src="test-file-stub"
              />
            </button>
          </div>
          <nav
            className="js-disabled-block"
          >
            <ul>
              <li>
                <a
                  href="/test-a"
                >
                  Test A
                </a>
              </li>
              <li>
                <a
                  href="/test-b"
                >
                  Test B
                </a>
              </li>
            </ul>
          </nav>
          <a
            className="getInvolved"
            href="/about-project/help"
          >
            Get Involved
          </a>
          <div
            className="mobilePointer"
          />
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "background": "colorPlum800",
            "height": "auto",
            "opacity": 1,
            "overflow": "unset",
            "transition": "height 500ms ease-in-out,opacity 500ms ease-in-out,background 500ms ease-in-out",
          }
        }
      >
        <div
          className="banner"
        >
          <span
            className="new"
          >
            New
          </span>
           
          <a
            href="/race"
          >
            We just launched the COVID Racial Data Tracker
          </a>
        </div>
      </div>
      <div
        className="container false container"
      >
        <div
          className="layout--content-full"
        >
          <div
            className="layout--content-primary"
          >
            <div
              className="siteTitleContainer"
            >
              <div
                className="siteTitleInner"
              >
                <a
                  href="/"
                >
                  <img
                    alt="The COVID Tracking Project"
                    src="test-file-stub"
                    width="176px"
                  />
                </a>
              </div>
              <div
                className="siteNavContainer"
              >
                <div
                  className="navContainer"
                >
                  <button
                    aria-expanded={false}
                    className="mobileToggle"
                    onClick={[Function]}
                    type="button"
                  >
                    Menu
                  </button>
                </div>
                <div
                  className="tools"
                >
                  <div
                    className="searchInput"
                  >
                    <div
                      data-reach-combobox=""
                    >
                      <input
                        aria-autocomplete="both"
                        aria-controls="listbox--1"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        data-reach-combobox-input=""
                        id="header-search"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        placeholder="Search"
                        role="combobox"
                        value=""
                      />
                    </div>
                    <button
                      className="searchSubmit"
                      onClick={[Function]}
                      type="button"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="test-file-stub"
                      />
                    </button>
                  </div>
                  <a
                    className="getInvolved"
                    href="/about-project/help"
                  >
                    Get involved
                  </a>
                </div>
                <nav
                  className="js-disabled-block"
                >
                  <ul>
                    <li>
                      <a
                        href="/test-a"
                      >
                        Test A
                      </a>
                    </li>
                    <li>
                      <a
                        href="/test-b"
                      >
                        Test B
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
            <div
              className="atlanticBanner"
            >
              <span>
                From
              </span>
               
              <img
                alt="The Atlantic"
                src="test-file-stub"
              />
              <div />
            </div>
            <div
              className="titleSubnavContainer"
            >
              <div
                className="title"
              >
                <h1
                  className="page-title pageTitle extraSpace"
                >
                  Sample title
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>,
  <main
    id="main"
  >
    <div
      data-reach-skip-nav-content=""
      id="reach-skip-nav"
    />
    <div
      className="container false container"
    >
      <div
        className="layout--content-full"
      >
        <div
          className="layout--content-primary"
        >
          <p>
            Content
          </p>
        </div>
      </div>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="container false container"
    >
      <div
        className="layout--content-full"
      >
        <div
          className="layout--content-primary"
        >
          <div
            className="container"
          >
            <a
              href="/"
            >
              <img
                alt="The COVID Tracking Project"
                src="test-file-stub"
              />
            </a>
            <ul
              className="footerList"
            >
              <li>
                <a
                  href="/contact"
                >
                  Contact
                </a>
              </li>
              <li>
                <a
                  href="/accessibility"
                >
                  Accessibility
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/COVID19Tracking"
                >
                  GitHub
                </a>
              </li>
              <li>
                <a
                  href="https://twitter.com/COVID19Tracking"
                >
                  Twitter
                </a>
              </li>
              <li>
                <a
                  className="backToTop"
                  href="#reach-skip-nav"
                >
                  Back to top.
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`Components : Layout : Header renders correctly 2`] = `
Array [
  <a
    data-reach-skip-link=""
    data-reach-skip-nav-link=""
    href="#reach-skip-nav"
  >
    Skip site navigation
  </a>,
  <header
    className="site-header siteHeader  "
  >
    <div
      className="container container showBackground "
    >
      <div
        className=""
        style={
          Object {
            "height": 0,
            "opacity": 0,
            "overflow": "hidden",
            "transition": "height 500ms ease-in-out,opacity 500ms ease-in-out,background 500ms ease-in-out",
          }
        }
      >
        <div
          className="mobileMenu"
        >
          <div
            className="searchInput"
          >
            <form
              action="/search"
              method="get"
              onSubmit={[Function]}
            >
              <input
                autoComplete="off"
                name="q"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Search"
                type="search"
              />
            </form>
            <button
              className="searchSubmit"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                aria-hidden="true"
                src="test-file-stub"
              />
            </button>
          </div>
          <nav
            className="js-disabled-block"
          >
            <ul>
              <li>
                <a
                  href="/test-a"
                >
                  Test A
                </a>
              </li>
              <li>
                <a
                  href="/test-b"
                >
                  Test B
                </a>
              </li>
            </ul>
          </nav>
          <a
            className="getInvolved"
            href="/about-project/help"
          >
            Get Involved
          </a>
          <div
            className="mobilePointer"
          />
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "background": "colorPlum800",
            "height": "auto",
            "opacity": 1,
            "overflow": "unset",
            "transition": "height 500ms ease-in-out,opacity 500ms ease-in-out,background 500ms ease-in-out",
          }
        }
      >
        <div
          className="banner"
        >
          <span
            className="new"
          >
            New
          </span>
           
          <a
            href="/race"
          >
            We just launched the COVID Racial Data Tracker
          </a>
        </div>
      </div>
      <div
        className="container false container"
      >
        <div
          className="layout--content-full"
        >
          <div
            className="layout--content-primary"
          >
            <div
              className="siteTitleContainer"
            >
              <div
                className="siteTitleInner"
              >
                <a
                  href="/"
                >
                  <img
                    alt="The COVID Tracking Project"
                    src="test-file-stub"
                    width="176px"
                  />
                </a>
              </div>
              <div
                className="siteNavContainer"
              >
                <div
                  className="navContainer"
                >
                  <button
                    aria-expanded={false}
                    className="mobileToggle"
                    onClick={[Function]}
                    type="button"
                  >
                    Menu
                  </button>
                </div>
                <div
                  className="tools"
                >
                  <div
                    className="searchInput"
                  >
                    <div
                      data-reach-combobox=""
                    >
                      <input
                        aria-autocomplete="both"
                        aria-controls="listbox--3"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        data-reach-combobox-input=""
                        id="header-search"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        placeholder="Search"
                        role="combobox"
                        value=""
                      />
                    </div>
                    <button
                      className="searchSubmit"
                      onClick={[Function]}
                      type="button"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="test-file-stub"
                      />
                    </button>
                  </div>
                  <a
                    className="getInvolved"
                    href="/about-project/help"
                  >
                    Get involved
                  </a>
                </div>
                <nav
                  className="js-disabled-block"
                >
                  <ul>
                    <li>
                      <a
                        href="/test-a"
                      >
                        Test A
                      </a>
                    </li>
                    <li>
                      <a
                        href="/test-b"
                      >
                        Test B
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
            <div
              className="atlanticBanner"
            >
              <span>
                From
              </span>
               
              <img
                alt="The Atlantic"
                src="test-file-stub"
              />
              <div />
            </div>
            <div
              className="titleSubnavContainer"
            >
              <div
                className="title"
              >
                <h1
                  className="page-title pageTitle extraSpace"
                >
                  Sample title
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>,
  <main
    id="main"
  >
    <div
      data-reach-skip-nav-content=""
      id="reach-skip-nav"
    />
    <div
      className="container textHeavy container"
    >
      <div
        className="layout--content-and-rail"
      >
        <div
          className="layout--content-primary"
        >
          <p>
            Content
          </p>
        </div>
      </div>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="container false container"
    >
      <div
        className="layout--content-full"
      >
        <div
          className="layout--content-primary"
        >
          <div
            className="container"
          >
            <a
              href="/"
            >
              <img
                alt="The COVID Tracking Project"
                src="test-file-stub"
              />
            </a>
            <ul
              className="footerList"
            >
              <li>
                <a
                  href="/contact"
                >
                  Contact
                </a>
              </li>
              <li>
                <a
                  href="/accessibility"
                >
                  Accessibility
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/COVID19Tracking"
                >
                  GitHub
                </a>
              </li>
              <li>
                <a
                  href="https://twitter.com/COVID19Tracking"
                >
                  Twitter
                </a>
              </li>
              <li>
                <a
                  className="backToTop"
                  href="#reach-skip-nav"
                >
                  Back to top.
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>,
]
`;
